Add "flat" support to loader

fdr opened the pull request:
In the case of models, I think it'd be useful to load all constants
into a flat, topmost namespace even if the files are organized in a
directory format.

There are two motivators:

1. Sequel requires specifying "class:" keywords onto every association
   if the target is not in the same module, e.g.:

        one_to_one :strand, key: :id, class: Strand

    It would be nice not to have to specify this all the time.

2. It encourages repetitive identifiers, e.g.

        class Postgres::PostgresServer

   To name this `Postgres::Server`, causes Sequel's to presume the
   database table name is also `server`

Both issues can be resolved by telegraphing information to Sequel in a
few annoying places per occurrence, but why bother?  I think it's
enough that directories allow running `git log $subdirectory` or
viewing `git diff --stat` and see what's going on.  It's not strictly
necessary they organize into modules/namespaces in Ruby.

So, with that in mind, load `models` differently, with the new `flat`
flag that ignores directories as it comes to module
creation/namespaces.

Example:

    $ mkdir model/test
    $ echo 'class Nested; end' > model/test/nested.rb
    $ ./bin/pry
    > Nested
    Nested

fdr commented:
@byucesoy the commit messages are written in the hypothetical where these changes did, in fact, help..but you should make sure they do first :)

byucesoy commented:
I rebased my branch on top of this and replaced "Postgres::PostgresServer" as "PostgresServer". It worked without any problem. Thanks for looking into this.

fdr commented:
> I rebased my branch on top of this and replaced "Postgres::PostgresServer" as "PostgresServer". It worked without any problem. Thanks for looking into this.

Were you able to delete all the class specifications, per rationale (1)?

If it seems pretty good to you, I guess I'll merge it. Cost is low, as we don't have any subdirectories of `model` just yet

byucesoy commented:
> Were you able to delete all the class specifications, per rationale (1)?

Yes, after this commit, I didn't need any class specificatoins.
